ngraph_bridge/ngraph_builder.cc

```diff
@@ -573,28 +563,53 @@ static Status TranslateArgMinMaxOp(
 
   if (tf_dim.size() != 1) {
     return errors::InvalidArgument(
-        (is_argmin ? "ArgMin" : "ArgMax"),
-        " Op: dimension must be scalar, operates on a single axis");
+        "ArgMax Op: dimension must be scalar, operates on a single axis");
   }
 
   // If input dimension is negative, make it positive
   if (tf_dim[0] < 0) {
+    NGRAPH_VLOG(3) << "Input dimension is negative, make it positive "
+                   << tf_dim[0];
     tf_dim[0] = (int64)input_rank + tf_dim[0];
   }
-  size_t input_dims = tf_dim[0];
+  NGRAPH_VLOG(3) << "Axis along which to compute " << tf_dim[0];
+  size_t k_axis = tf_dim[0];
 
   DataType dtype;
   TF_RETURN_IF_ERROR(GetNodeAttr(op->attrs(), "output_type", &dtype));
 
   ng::element::Type ng_et;
   TF_RETURN_IF_ERROR(TFDataTypeToNGraphElementType(dtype, &ng_et));
 
-  auto ng_argminmax =
-      ConstructNgNode<T>(op->name(), ng_input, input_dims, ng_et);
-  SaveNgOp(ng_op_map, op->name(), ng_argminmax);
+  auto ng_k = ConstructNgNode<opset::Constant>(
+      op->name(), ng::element::i64, ng::Shape{}, std::vector<int>({1}));
+
+  std::string sort = "none";
+  auto ng_topk =
+      std::make_shared<opset::TopK>(ng_input, ng_k, k_axis, mode, sort, ng_et);
+  auto ng_indices = ng_topk->output(1);
+  int axis = ng_topk->get_axis();
+  auto axis_to_remove = ConstructNgNode<opset::Constant>(
+      op->name(), ng::element::i64, ng::Shape{}, std::vector<int>({axis}));
```

Comment:
Use the same type here as the nGraph element type. Also, does aggregate initialization work here or do you have to use the `vector` constructor?

Finally, I think it'd be better to be explicit about the shape here.